wgengine/router/router_linux.go

```diff
@@ -92,40 +92,38 @@ func chooseFireWallMode(logf logger.Logf, det tableDetector) (linuxfw.FirewallMo
 	case envknob.String("TS_DEBUG_FIREWALL_MODE") == "nftables":
 		// TODO(KevinLiang10): Updates to a flag
 		logf("router: envknob TS_DEBUG_FIREWALL_MODE=nftables set")
-		return linuxfw.FirewallModeNfTables, nil
+		return linuxfw.FirewallModeNfTables
 	case envknob.String("TS_DEBUG_FIREWALL_MODE") == "iptables":
 		logf("router: envknob TS_DEBUG_FIREWALL_MODE=iptables set")
-		return linuxfw.FirewallModeIPTables, nil
+		return linuxfw.FirewallModeIPTables
 	case nftRuleCount > 0 && iptRuleCount == 0:
 		logf("router: nftables is currently in use")
-		return linuxfw.FirewallModeNfTables, nil
+		return linuxfw.FirewallModeNfTables
 	case iptRuleCount > 0 && nftRuleCount == 0:
 		logf("router: iptables is currently in use")
-		return linuxfw.FirewallModeIPTables, nil
+		return linuxfw.FirewallModeIPTables
 	case nftAva:
 		// if both iptables and nftables are available but
 		// neither/both are currently used, use nftables.
 		logf("router: nftables is available")
-		return linuxfw.FirewallModeNfTables, nil
+		return linuxfw.FirewallModeNfTables
 	case iptAva:
 		logf("router: iptables is available")
-		return linuxfw.FirewallModeIPTables, nil
+		return linuxfw.FirewallModeIPTables
 	default:
 		// if neither iptables nor nftables are available,
```

Comment:
Please update this comment to describe what this behavior is and why we have this behavior (essentially what you have in the commit message)